src/back/wgsl/writer.rs

```diff
@@ -917,11 +918,26 @@ impl<W: Write> Writer<W> {
                     self.write_stmt(module, sta, func_ctx, l2)?;
                 }
 
-                if !continuing.is_empty() {
+                // The continuing is optional so we don't need to write it if
+                // it is empty, but the `break if` counts as a continuing statement
+                // so even if `continuing` is empty we must generate it if a
+                // `break if` exists
```

Comment:
I have been implementing your suggestion and honestly I think it's more effort than it's worth it, because now everytime you encounter a continuing block you need to match on it, access the body and continue your work

Reply:
Okay, thanks for trying it out. I'm fine with this as originally posted.